plugins/flutter_aepcore/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepcore/FlutterAEPCoreDataBridge.java

```diff
@@ -127,4 +162,27 @@ private static String getNullableString(final Map data, final String key) {
     private static Map getNullableMap(final Map data, final String key) {
         return data.containsKey(key) && (data.get(key) instanceof Map) ? (Map) data.get(key) : null;
     }
+
+    private static Boolean getNullableBoolean(final Map data, final String key) {
+        return data.containsKey(key) && (data.get(key) instanceof Boolean) ? (Boolean) data.get(key) : null;
+    }
+
+    private static Map<String, String> getNullableStringMap(final Map data, final String key) {
+    if (!data.containsKey(key) || !(data.get(key) instanceof Map)) {
+        return null;
+    }
+
+    Map rawMap = (Map) data.get(key);
+    Map<String, String> stringMap = new HashMap<>();
```

Comment:
Don't create a new map. Instead, after validation, your return value on line 186 should be (Map<String, String>) rawMap.